test/server/http/admin_test.cc

```diff
@@ -68,13 +84,27 @@ TEST_F(AdminFilterTest, AdminProfiler) {
 
 #endif
 
-TEST_F(AdminFilterTest, AdminBadProfiler) {
+TEST_F(AdminInstanceTest, AdminBadProfiler) {
   Buffer::OwnedImpl data;
-  AdminImpl admin_bad_profile_path("/dev/null",
-                                   TestEnvironment::temporaryPath("some/unlikely/bad/path.prof"),
-                                   Network::Utility::resolveUrl("tcp://127.0.0.1:0"), server_);
+  AdminImpl admin_bad_profile_path(
+      "/dev/null", TestEnvironment::temporaryPath("some/unlikely/bad/path.prof"), "",
```

Comment:
Yeah, I think this points to making `AdminImplOptions` something worth considering.